### PR TITLE
Implement isolate `remove_slot`

### LINKED ISSUE #768: Support clearing an Isolate slot
There are `get_slot`, `get_mut_slot` and `set_slot` methods, they are very nice.

But it could be a very nice improvement if it made possible to get the value back out of the slot as owned, and clear the slots content in the operation.

A `get_slot_owned` method, which returns the value itself clearing the slot.

Instead of returning a reference.

I could try and do a PR for this if it's technically possible!

### PR DESCRIPTION
**Pull request by Rami-Sabbagh**
This should resolve #768 

Documentation and a test unit have written, hope it's a good quality PR!

### CODE REVIEW

**Comment by Rami-Sabbagh** [comment]
Fixed the formatting issue which caused the tests to fail

File: tests/slots.rs [hunk 1]
```diff
@@ -163,3 +163,20 @@ fn slots_layer2() {
...
+    core_isolate.remove_slot::<CoreIsolateState>().unwrap();
+  drop(core_isolate);
+
+  // The state should not be dropped with the isolate because we own it now.
+  assert_eq!(drop_count.load(Ordering::SeqCst), 0);
+
+  // We're dropping it now on purpose.
+  drop(state);
+  assert_eq!(drop_count.load(Ordering::SeqCst), 1);
+}
```
Comment (bartlomieju) [inline tests/slots.rs:182]: Can you rework the test a bit to first add struct to a slot, increment it, remove the slot and verify that following calls to get the slot fail?

Comment (Rami-Sabbagh) [reply tests/slots.rs:182]: Sure, give me a few minutes

**Comment by Rami-Sabbagh** [comment]
Oooof it's gonna keep complaining, will check the actual CI workflow to figure out what other checks are there

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @Rami-Sabbagh cleanly done!

**Comment by Rami-Sabbagh** [comment]
Really thanks for reviewing the PR 😄 
It means a lot for me to have a contribution on such a project.

**Pull request merged**